homeassistant/components/aprilaire/config_flow.py

```diff
@@ -0,0 +1,74 @@
+"""Config flow for the Aprilaire integration."""
+
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_HOST, CONF_PORT
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import DOMAIN, LOG_NAME
+from .coordinator import AprilaireCoordinator
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST): str,
+        vol.Required(CONF_PORT, default=7000): int,
+    }
+)
+
+_LOGGER = logging.getLogger(LOG_NAME)
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Aprilaire."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        if user_input is None:
+            return self.async_show_form(
+                step_id="user", data_schema=STEP_USER_DATA_SCHEMA
+            )
+
+        errors = {}
+
+        try:
+            await self.async_set_unique_id(
+                f'aprilaire_{user_input[CONF_HOST].replace(".", "")}{user_input[CONF_PORT]}'
+            )
+            self._abort_if_unique_id_configured()
+        except Exception as err:  # pylint: disable=broad-except
+            _LOGGER.exception("Unexpected exception")
+            errors["base"] = str(err)
+        else:
+            coordinator = AprilaireCoordinator(
```

Comment:
I still need to see what the coordinator does, but I don't prefer to call the coordinator in the config flow

Reply:
@joostlek It's probably not ideal, but the complication comes from the `wait_for_ready` method. Due to the asynchronous nature of the connection and it being pushed based, connecting requires a multi-step process, which is what that method does. Basically, we need to wait for the data to be in the coordinator, regardless of if that is because we sent a request for it or the thermostat sent it on its own. I was thinking if it would be possible to shift the `wait_for_ready` method to the client (which is in the `pyaprilaire` package), but there is a dependence on the coordinator's state to make it work.

Considering that, I think there are two options:

1. Leave it as-is, favoring properly testing the connection at the expense of referencing the coordinator in the config flow
2. Make the config flow only check that the socket is accessible at the host/port combination, and not check that anything else about the connection is successful.

I think the user experience of the first option is better, avoiding adding a thermostat that doesn't actually connect, which does happen. But I welcome your thoughts.